File: mianyang-campus/backend/app/models/feedback.py
from datetime import datetime
from sqlalchemy import String, Text, Boolean, DateTime, ForeignKey, Enum
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy.sql import func
from app.core.database import Base
import enum


class FeedbackType(str, enum.Enum):
    BUG = "bug"
    FEATURE = "feature"
    COMPLAINT = "complaint"
    OTHER = "other"


class FeedbackStatus(str, enum.Enum):
    PENDING = "pending"
    PROCESSING = "processing"
    RESOLVED = "resolved"
    REJECTED = "rejected"


class Feedback(Base):
    __tablename__ = "feedbacks"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), nullable=False, index=True)
    type: Mapped[FeedbackType] = mapped_column(Enum(FeedbackType), default=FeedbackType.OTHER)
    title: Mapped[str] = mapped_column(String(200), nullable=False)
    content: Mapped[str] = mapped_column(Text, nullable=False)
    contact: Mapped[str | None] = mapped_column(String(100), nullable=True)
    status: Mapped[FeedbackStatus] = mapped_column(Enum(FeedbackStatus), default=FeedbackStatus.PENDING, index=True)
    reply: Mapped[str | None] = mapped_column(Text, nullable=True)
    replied_by: Mapped[int | None] = mapped_column(ForeignKey("users.id"), nullable=True)
    replied_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())

    user = relationship("User", foreign_keys=[user_id], back_populates="feedbacks")
    replier = relationship("User", foreign_keys=[replied_by])

File: mianyang-campus/frontend/src/api/feedback.ts
import request from '@/utils/request'

export interface Feedback {
  id: number
  user_id: number
  user_name?: string
  type: string
  title: string
  content: string
  contact?: string
  status: string
  reply?: string
  replied_by?: number
  replier_name?: string
  replied_at?: string
  created_at: string
}

export interface FeedbackCreate {
  type?: string
  title: string
  content: string
  contact?: string
}

export interface FeedbackReply {
  reply: string
  status?: string
}

// 提交反馈
export function createFeedback(data: FeedbackCreate) {
  return request.post<Feedback>('/feedbacks', data)
}

// 获取反馈列表
export function getFeedbacks(params?: {
  status?: string
  type?: string
  page?: number
  page_size?: number
}) {
  return request.get<Feedback[]>('/feedbacks', { params })
}

// 获取反馈详情
export function getFeedback(id: number) {
  return request.get<Feedback>(`/feedbacks/${id}`)
}

// 回复反馈（管理员）
export function replyFeedback(id: number, data: FeedbackReply) {
  return request.put<Feedback>(`/feedbacks/${id}/reply`, data)
}
